src/components/ModalDrawer.js

```diff
@@ -0,0 +1,131 @@
+    display: 'flex', flexDirection: 'column'
+  }),
+  title: {
+    display: 'flex', alignItems: 'baseline', marginBottom: '1rem', flex: 'none', padding: '1.5rem 1.25rem'
+  },
+  body: {
+    display: 'flex', alignItems: 'baseline', marginBottom: '1rem', flex: 1, padding: '1.5rem 1.25rem', overflow: 'auto'
+  },
+  buttonRow: {
+    marginTop: 'auto', backgroundColor: colors.dark(0.2), padding: '1.75rem 1.25rem',
+    display: 'flex', justifyContent: 'flex-end', alignItems: 'baseline'
+  },
+  titleAlign: showPrevArrow => ({ marginLeft: showPrevArrow ? 'auto' : undefined })
+}
+
+const Modal = ({
+  onPrevious = () => {}, showPreviousArrow = false, transitionState = false,
+  ...props
+}) => {
+  const { onDismiss, title, titleExtras, children, width, showCancel, cancelText, showX, showButtons, okButton, ...restProps } = props
+
+  return h(RModal, {
+    parentSelector: () => document.getElementById('modal-root'),
+    isOpen: true,
+    onRequestClose: onDismiss,
+    style: { overlay: drawer.overlay(transitionState), content: { ...drawer.container(transitionState), width } },
+    ariaHideApp: false,
+    ...restProps
+  }, [
+    title && div({ style: drawer.title }, [
+      showPreviousArrow && h(Clickable, {
+        onClick: onPrevious
+      }, [icon('arrowLeft')]),
+      div({ style: { fontSize: 18, fontWeight: 600, ...drawer.titleAlign(showPreviousArrow) } }, [title]),
+      titleExtras,
+      showX && h(Clickable, {
+        style: { marginLeft: 'auto' },
+        onClick: onDismiss
+      }, [icon('times')])
+    ]),
+    div({ style: drawer.body }, [children]),
+    showButtons && div({ style: drawer.buttonRow }, [
+      showCancel ?
+        buttonSecondary({
+          style: { marginRight: '3rem' },
+          onClick: onDismiss
+        }, [cancelText]) :
+        null,
+      Utils.cond(
+        [okButton === undefined, () => buttonPrimary({ onClick: onDismiss }, 'OK')],
+        [_.isString(okButton), () => buttonPrimary({ onClick: onDismiss }, okButton)],
+        [_.isFunction(okButton), () => buttonPrimary({ onClick: okButton }, 'OK')],
+        () => okButton
+      )
+    ])
+  ])
+}
+
+const propTypes = {
+  onDismiss: PropTypes.func.isRequired,
+  title: PropTypes.node,
+  titleExtras: PropTypes.node,
+  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+  showCancel: PropTypes.bool,
+  cancelText: PropTypes.string,
+  showX: PropTypes.bool,
+  showButtons: PropTypes.bool,
+  okButton: PropTypes.oneOfType([PropTypes.string, PropTypes.func, PropTypes.node]),
+  children: PropTypes.node
+}
+
+const defaultProps = {
+  width: 450,
+  showCancel: true,
+  cancelText: 'Cancel',
+  showX: false,
+  showButtons: false,
+  showPreviousArrow: false,
+  openDrawer: false
+}
+
+export const ModalDrawer = props => {
+  const { openDrawer, onDismiss } = props
+  const [closeDrawer, setCloseDrawer] = useState(false)
+
+  const transitionModal = h(Transition, {
+    in: openDrawer && !closeDrawer,
```

Comment:
This setup will make it difficult for the parent to close the modal programmatically by clearing a boolean. With the current usage pattern, that would immediately unmount the drawer and skip the transition. If we definitely want to have an exit transition, it will probably be cleaner to require the drawer to stay mounted, and control the visibility via an `isOpen` prop. That way we can hide the timing details of the transition, and expose a simpler interface to the parent.